### ISSUE TITLE
Millions of `/query/profile` requests

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#12921]
Over the last hour (in which I did rejoin `#element:matrix.org`, but still), `matrix.org` has sent my server 1892 `GET /_matrix/federation/v1/query/profile?user_id=%40richvdh%3Asw1v.org` requests.

They aren't an inherent problem, since they can be served pretty quickly, but this still smacks of huge inefficiency somewhere.

**Comment by DMRobertson** [comment]
(That's a surprise. I could've sworn the remote profile cache was somehow related to the user directory. Apparently not.)

**Comment by turt2live** [comment]
I posted a message to HQ (2 if you count edits) and so far I'm at over 1500 requests all of a sudden.

```
# cat federation_reader_1.log | grep "matrix.org.*query/profile" | wc -l
1650
```

This sounds like it's related to matrix.org trying to send out push/email notifications or something, failing to cache the sender between notifications. 

Currently this is an annoyance on the receiver side, though surely on the matrix.org/remote side it'd be burning a small hole in a bandwidth budget somewhere?

**Comment by richvdh** [comment]
> This sounds like it's related to matrix.org trying to send out push/email notifications or something, failing to cache the sender between notifications.

no, it's from matrix.org users on Element Android making `GET /_matrix/client/r0/profile/<userid>` requests, each of which triggers *two* `GET /_matrix/federation/v1/query/profile` requests, because (a) there is no caching whatsoever (b) displayname and avatar are fetched separately.

Some sort of caching seems like it would be helpful.
